exec/java-exec/src/main/java/org/apache/drill/exec/service/ServiceEngine.java

```diff
@@ -102,6 +105,7 @@ public DrillbitEndpoint start() throws DrillbitStartupException, UnknownHostExce
     DrillbitEndpoint partialEndpoint = DrillbitEndpoint.newBuilder()
         .setAddress(hostName)
         .setUserPort(userPort)
+        .setHttpPort(httpPort)
```

Comment:
Recently port-hunting capability is added to WebServer. So I think we should get port number from the WebServer.

Reply:
That's tricky. The webserver doesn't come up until after the Endpoint is created and ready. One possible solution would be to perform the port hunt before the endpoint, and assign it to a newly instantiated webserver after the Endpoint is created.

Reply:
Thanks to @dvjyothsna , referred to an outdated PR #1042 for the endpoint registration after the web-server starts up.